src/engine/source/kvdb/include/kvdb/refCounter.hpp

```diff
@@ -0,0 +1,92 @@
+#ifndef _REF_COUNTER_H
+#define _REF_COUNTER_H
+
+#include <map>
+#include <string>
+#include <vector>
+
+namespace kvdbManager
+{
+
+/**
+ * @brief Basic Helper wrapping std::map<std::string, int> to count references.
+ *
+ */
+class RefCounter
+{
+public:
+    /**
+     * @brief Construct a new RefCounter object
+     *
+     */
+    RefCounter() = default;
+
+    /**
+     * @brief Construct a new RefCounter object by copy.
+     *
+     * @param refMap object to copy.
+     *
+     */
+    RefCounter(const RefCounter&);
+
+    /**
+     * @brief Destroy the RefCounter object
+     *
+     */
+    ~RefCounter() { m_refMap.clear(); }
+
+    /**
+     * @brief Add n reference(s) to the counter.
+     *
+     * @param name Reference name to increase
+     * @param times How many times to increase the reference. Default is 1.
+     */
+    void addRef(const std::string& name, const unsigned int& times = 1);
```

Comment:
Always pass trivial types by value.

Also using uint to increase an int.

Reply:
I'm sorry, but there seems to be a misunderstanding. What I meant about the unsigned int is that the map stores an int and we use an unsigned type to increase said int. Therefore performing a narrowing conversion.

This kind of error should be warned by clang-tidy, which is integrated into vs code:
![image](https://github.com/wazuh/wazuh/assets/32262972/d8ae7043-f4ad-4177-9791-39a2d606948d)

Reply:
Fair enough observation Javier. Fixed. Thank you.